cmd/forward_test.go

```diff
@@ -46,6 +46,31 @@ func TestParseArgs(t *testing.T) {
 	})
 }
 
+type SafeBuffer struct {
+	mutex sync.RWMutex
```

Comment:
The use of a mutex implies that you expect multiple goroutines to write to this buffer. This is often an anti-pattern as routines should pass data around via channels rather than directly sharing memory:

https://go.dev/blog/codelab-share

Reply:
See also https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions#NewTestIOStreams

Did you get a data race here?

Reply:
I got a data race yeah

Reply:
We only share the buffer between commands in the test here because we're subbing out the os.Stdout/Stderr to check the outputs. A mutex was the first thing that came to mind, if there is a better approach that fits with the passed streams API then I'm for it. I would assume the same problem exists with the test IO streams because they are also just buffers

Reply:
You could handle collecting output from multiple routines via channels but it's probably not worth it. 

However, you should accomplish this by wrapping a `bytes.Buffer` with mutex locking rather than reimplementing it with `[]byte`.